x-pack/plugins/apm/public/components/shared/charts/transaction_charts/helper.tsx

```diff
@@ -23,11 +24,14 @@ export function getResponseTimeTooltipFormatter(formatter: TimeFormatter) {
   };
 }
 
-export function getMaxY(timeSeries?: Array<TimeSeries<Coordinate>>) {
-  if (timeSeries) {
-    const coordinates = timeSeries.flatMap((serie) => serie.data);
-    const numbers = coordinates.map((c) => (c.y ? c.y : 0));
-    return Math.max(...numbers, 0);
+export function getMaxY(specs?: Array<APMChartSpec<Coordinate>>) {
+  const values = specs
+    ?.flatMap((spec) => spec.data)
+    .map((coord) => coord.y)
+    .filter(isFiniteNumber);
```

Comment:
Can you drop the second map?

```suggestion
    ?.flatMap((spec) => spec.data.y)
    .filter(isFiniteNumber);
```

Reply:
How would this work? `spec.data` is an array of coordinates. `flatMap` only flattens one level.

Reply:
ie, `y` is not a property of `data`, but of `data[0]`

Reply:
Ah, obviously. I was reading it wrong.